lls/main.c

```diff
@@ -316,6 +317,124 @@ lacp_timer_reset(struct lls_config *lls_conf, struct gatekeeper_if *iface)
 		lls_conf->lcore_id, lls_lacp_announce, iface);
 }
 
+static void
+fillup_lls_dump_entry(struct lls_dump_entry *dentry, struct lls_map *map,
+	uint32_t key_len)
+{
+	dentry->stale = map->stale;
+	dentry->port_id = map->port_id;
+	if (key_len == sizeof(struct in_addr)) {
+		dentry->ipaddr.proto = ETHER_TYPE_IPv4;
+		rte_memcpy(&dentry->ipaddr.ip.v4, map->ip_be,
+			sizeof(dentry->ipaddr.ip.v4));
```

Comment:
Doesn't `dentry->ipaddr.ip.v4 = map->ip_be;` match types?

Reply:
Not exactly, we used `uint8_t ip_be[LLS_MAX_KEY_LEN]` as the key for both IPv4 and IPv6 LLS entries, before we devised `struct ipaddr`. Now it seems that the union of `struct in_addr` and `struct in6_addr` inside of `struct ipaddr` is used as they key for hash tables in the GK block, and could be adopted for use in the LLS block.

This would also allow us to drop the constant `LLS_MAX_KEY_LEN`. Should I add a patch to do this as part of this pull request?

Reply:
Thanks for doing it.